File: src/main/scala/br/fcv/wiquery_test/pjax/ListFilesPage.scala
package br.fcv.wiquery_test.pjax;

import org.apache.wicket.markup.html.IHeaderResponse
import org.apache.wicket.markup.html.WebPage
import org.apache.wicket.request.cycle.RequestCycle
import org.apache.wicket.request.http.WebRequest
import org.apache.wicket.request.mapper.parameter.PageParameters
import br.fcv.wiquery_test.support.wicket.JQueryPjaxJavaScriptReference
import org.odlabs.wiquery.core.resources.CoreJavaScriptResourceReference


class ListFilesPage(parameters: PageParameters) extends WebPage {

    add( contentPanel("content", new PageParameters(parameters).remove("_pjax")) )

    protected def isPjax = {
        val request = RequestCycle.get().getRequest().asInstanceOf[WebRequest];
        val pjax = request.getHeader("X-PJAX")
        pjax != null && pjax.toBoolean
    }
    
    override def renderHead(response: IHeaderResponse) {
        response.renderJavaScriptReference(CoreJavaScriptResourceReference.get());
        response.renderJavaScriptReference(JQueryPjaxJavaScriptReference.getInstance());
    }
    
    def contentPanel(id: String, params: PageParameters) = { 
        if (isPjax) { 
            new ListFilesPanel(id, params) 
        } else {
            new FullContentPanel(id, params)
        }
    }
}

File: src/main/java/br/fcv/wiquery_test/pjax/ListFilesPage.java
package br.fcv.wiquery_test.pjax;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

import org.apache.wicket.markup.html.WebMarkupContainer;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.model.LoadableDetachableModel;
import org.apache.wicket.request.cycle.RequestCycle;
import org.apache.wicket.request.http.WebRequest;
import org.apache.wicket.request.mapper.parameter.PageParameters;

public class ListFilesPage extends WebPage {
    
    public ListFilesPage(PageParameters parameters) {
        add(new Header("header"));
    }
    
    public boolean isPjax() {
        WebRequest request = (WebRequest) RequestCycle.get().getRequest();
        return Boolean.parseBoolean(request.getHeader("X-PJAX")); 
    }
    
    private static class Header extends WebMarkupContainer {

        public Header(String id) {
            super(id);
            add(new Label("now", new NowModel()));
        }
        
    }
    
    private static class NowModel extends LoadableDetachableModel<String> {
        
        public static final String DEFAULT_FORMAT = "dd-MM-yyyy HH:mm:ss.SSSS";
        
        private final String format;
        
        public NowModel() {
            this(DEFAULT_FORMAT);
        }
        
        public NowModel(String format) {
            this.format = format;
        }

        @Override
        protected String load() {
            DateFormat formatter = new SimpleDateFormat(format);
            return formatter.format(new Date());
        }
        
    }

}
